src/Interpreters/Context.cpp

```diff
@@ -2739,8 +2739,11 @@ void Context::setCurrentQueryId(const String & query_id)
 
     client_info.current_query_id = query_id_to_set;
 
-    if (client_info.query_kind == ClientInfo::QueryKind::INITIAL_QUERY)
+    if (client_info.query_kind == ClientInfo::QueryKind::INITIAL_QUERY
+        && (getApplicationType() != ApplicationType::SERVER || client_info.initial_query_id.empty()))
```

Comment:
I do not completely understand this change.
Why do we need `getApplicationType() != ApplicationType::SERVER` check?

Reply:
ClickHouse client reuses client_info structure in different requests, and in this place `initial_query_id` has value from previous request.
ClickHouse server recreates structure for each request, so I made this workaround to keep `initial_query_id` only in server, when it got some value from initiator.

May be not best solution.

I think that correct way is to make some flag for this case or another additional value like `SECONDARY_INITIAL_QUERY` for `query_kind`, but this variants break backward compatibility.

Also possible not to change behavior here, it does not break query result, but than this 'secondary initial' query gets new `initial_query_id`, and as result impossible to reassemble all subqueries from `system.query_log` for any kind of analyze.